homeassistant/components/planifneige/sensor.py

```diff
@@ -0,0 +1,135 @@
+"""Sensor for the City of Montreal's Planif-Neige snow removal APIs."""
+
+from datetime import datetime
+import logging
+
+from homeassistant.components.planifneige import (
+    CONF_STREETID, DATA_PLANIFNEIGE, PLANIFNEIGE_ATTRIBUTION)
+from homeassistant.const import ATTR_ATTRIBUTION, CONF_NAME
+from homeassistant.helpers.restore_state import RestoreEntity
+
+DEPENDENCIES = ['planifneige']
+
+_LOGGER = logging.getLogger(__name__)
+
+DB_TIMEFORMAT = '%Y-%m-%d %H:%M:%S'
+
+ATTR_STREET_SIDE_ID = 'street_side_id'
+ATTR_START_PLAN_DATE = 'start_plan_date'
+ATTR_END_PLAN_DATE = 'end_plan_date'
+ATTR_START_REPLAN_DATE = 'start_replan_date'
+ATTR_END_REPLAN_DATE = 'end_replan_date'
+ATTR_UPDATED = 'updated'
+
+STREET_STATE = {
+    '0': ['Snowed', 'mdi:snowflake'],  # street is snowed in
+    '1': ['Clear', 'mdi:road'],  # street is clear
+    '2': ['Planned', 'mdi:clock-outline'],  # street is planned for clearing
+    '3': ['Replanned', 'mdi:clock-alert-outline'],  # clearing date replanned
+    '4': ['Snowed', 'mdi:snowflake'],  # to be planned, still snowed in
+    '5': ['Clearing', 'mdi:bulldozer'],  # trucks loading, still clearing
+    '10': ['Ploughed', 'mdi:road']  # street cleared, snow not loaded
+}
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the PlanifNeige platform."""
+    data = hass.data[DATA_PLANIFNEIGE]
+    async_add_entities(
+        [PlanifNeigeSensor(data, sensor) for sensor in discovery_info]
+    )
+
+
+class PlanifNeigeSensor(RestoreEntity):
```

Comment:
If restore state should work we need to handle that in `async_added_to_hass`. See other platforms that use restore state for examples.

Reply:
We should not use restore state because the data can be fetched from an API.